source/Coop/Mod/Persistence/RPC/EventBroadcastingQueue.cs

```diff
@@ -98,15 +98,17 @@ public void Update(TimeSpan frameTime)
         /// </summary>
         /// <param name="room">Room to send the event to.</param>
         /// <param name="rpc"></param>
-        public void Add(RailServerRoom room, EventMethodCall rpc)
+        public bool Add(RailServerRoom room, EventMethodCall rpc)
         {
             lock (m_Queue)
             {
                 Call call = new Call(room, rpc);
                 call.ObjectsToBeDistributed.RemoveAll(id => m_DistributedObjects.Contains(id));
 
+                bool isFull = false;
                 if (m_Queue.Count >= MaximumQueueSize)
```

Comment:
The pause should happen before an event gets lost. The condition should probably be `if(m_Queue.Count >= (MaximumQueueSize - 1)`.